cmd/neofs-node/reputation/intermediate/consumers.go

```diff
@@ -0,0 +1,53 @@
+package intermediate
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/nspcc-dev/neofs-node/pkg/services/reputation"
+	"github.com/nspcc-dev/neofs-node/pkg/services/reputation/common"
+	reputationcommon "github.com/nspcc-dev/neofs-node/pkg/services/reputation/common"
+	"github.com/nspcc-dev/neofs-node/pkg/services/reputation/eigentrust"
+	eigencalc "github.com/nspcc-dev/neofs-node/pkg/services/reputation/eigentrust/calculator"
+	consumerstorage "github.com/nspcc-dev/neofs-node/pkg/services/reputation/eigentrust/storage/consumers"
+	"github.com/nspcc-dev/neofs-node/pkg/util/logger"
+)
+
+var ErrIncorrectContext = errors.New("could not write intermediate trust: passed context incorrect")
+
+type ConsumerStorageWriterProvider struct {
+	Log     *logger.Logger
+	Storage *consumerstorage.Storage
+}
+
+type ConsumerTrustWriter struct {
+	log     *logger.Logger
+	storage *consumerstorage.Storage
+}
+
+func (w *ConsumerTrustWriter) Write(ctx common.Context, t reputation.Trust) error {
+	eiCtx, ok := ctx.(eigencalc.Context)
+	if !ok {
+		return ErrIncorrectContext
```

Comment:
Seems not good, why do we restrict context?

Reply:
because intermediate `Trust`s have no sense without `I()` and `Epoch()` "context"

Reply:
So we try to implement more detailed interface through more abstracted one, don't you think this is incorrect?

Reply:
found no better solution

there must be always `Epoch()` method, but for intermediate `I()` is also required. and for local `Trust` `I()` means nothing and does not exist

i look at this solution in the same way as i look at ctx with `passedRoute` unnecessary field. i may be wrong